IdentityCore/src/cache/MSIDMacKeychainTokenCache.m

```diff
@@ -0,0 +1,658 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import <Foundation/Foundation.h>
+#import "MSIDAccountCacheItem.h"
+#import "MSIDAccountItemSerializer.h"
+#import "MSIDAccountType.h"
+#import "MSIDCredentialCacheItem.h"
+#import "MSIDCredentialItemSerializer.h"
+#import "MSIDCredentialType.h"
+#import "MSIDError.h"
+#import "MSIDJsonSerializer.h"
+#import "MSIDLogger+Internal.h"
+#import "MSIDLogger.h"
+#import "MSIDMacKeychainTokenCache.h"
+#import "MSIDUserInformation.h"
+#import "NSString+MSIDExtensions.h"
+#import "MSIDKeychainUtil.h"
+#import "MSIDMacKeychainAccountCacheKey.h"
+
+/**
+This Mac cache stores serialized account and credential objects in the macOS "login" Keychain.
+There are three types of items stored:
+  1) Secret shareable artifacts (SSO credentials: Refresh tokens, other global credentials)
+  2) Non-secret shareable artifacts (account metadata)
+  3) Secret non-shareable artifacts (access tokens, ID tokens)
+
+In addition to the basic account & credential properties, the following definitions are used below:
+<account_id>    : “<home_account_id>-<environment>”
```

Comment:
nit: the first three are aligned by `:` unlike the last one

Reply:
updated